pkgs/os-specific/linux/isgx/default.nix

```diff
@@ -0,0 +1,46 @@
+{ stdenv, lib, fetchFromGitHub, fetchpatch, kernel }:
+
+stdenv.mkDerivation rec {
+  name = "isgx-${version}-${kernel.version}";
+  version = "2.11";
+
+  src = fetchFromGitHub {
+    owner = "intel";
+    repo = "linux-sgx-driver";
+    rev = "sgx_driver_${version}";
+    hash = "sha256-zZ0FgCx63LCNmvQ909O27v/o4+93gefhgEE/oDr/bHw=";
+  };
+
+  patches = [
+    # Fixes build with kernel >= 5.8
+    (fetchpatch {
+      url = "https://github.com/intel/linux-sgx-driver/commit/276c5c6a064d22358542f5e0aa96b1c0ace5d695.patch";
+      sha256 = "sha256-PmchqYENIbnJ51G/tkdap/g20LUrJEoQ4rDtqy6hj24=";
+    })
+  ];
+
+  hardeningDisable = [ "pic" ];
+
+  nativeBuildInputs = kernel.moduleBuildDependencies;
+
+  makeFlags = [
+    "KDIR=${kernel.dev}/lib/modules/${kernel.modDirVersion}/build"
+    "ARCH=${stdenv.hostPlatform.linuxArch}"
+  ] ++ lib.optional (stdenv.hostPlatform != stdenv.buildPlatform) [
+    "CROSS_COMPILE=${stdenv.cc.targetPrefix}"
+  ];
+
+  installPhase = ''
+    runHook preInstall
+    install -Dm644 isgx.ko -t $out/lib/modules/${kernel.modDirVersion}/kernel/drivers/intel/sgx
+    runHook postInstall
+  '';
+
+  meta = with lib; {
+    description = "Intel SGX Linux Driver";
+    homepage = "https://github.com/intel/linux-sgx-driver";
+    license = with licenses; [ bsd3 gpl2 ];
```

Comment:
I am inclined to say we no longer need this package because current linux testing adds an upstream sgx driver. Nevertheless you should mark this as broken for `5.11.0`

Reply:
Btw this is what you need to enable to have it in the kernel: https://github.com/Mic92/doctor-cluster-config/commit/bb70ed032ec876b93e2c4263af3542aa56fd7bdd#diff-d33b7b1f936a6f8f74abe312c8ef8737c4b53a525878f08564a4d0c163f2be5cR15

Reply:
also gpl2Only or gpl2Plus?

Reply:
@Mic92 Broken predicate is added. BTW: Kernel 5.11 is not in master currently. I'll check it later.

@SuperSandro2000 It's gpl2Only. Fixed.